src/haxelib/api/Vcs.hx

```diff
@@ -188,30 +164,93 @@ class Vcs implements IVcs {
 		return available;
 	}
 
-	@:final function get_available():Bool {
+	final function get_available():Bool {
 		if (!availabilityChecked)
 			checkExecutable();
 		return available;
 	}
 
-	public function clone(libPath:String, vcsPath:String, ?branch:String, ?version:String):Void {
-		throw "This method must be overridden.";
+	final function runStrict(args:Array<String>, ?debugLog:(msg:String) -> Void):{
+		code:Int,
+		out:String,
+		err:String
+	} {
+		inline function print(msg)
+			if (debugLog != null && msg != "")
+				debugLog(msg);
+
+		print("# Running command: " + executable + " " + args.toString() + "\n");
+
+		final proc = command(executable, args);
+		switch (proc) {
+			case {code: 0}: // pass
+				print(proc.out);
+				print(proc.err);
+				print('# Exited with code ${proc.code}\n');
+			case {code: code, out: out, err: err}:
+				throw CommandFailed(this, code, out, err);
+		}
+		return proc;
+	}
+
+	final function run(args:Array<String>, ?debugLog:(msg:String) -> Void):{
```

Comment:
`run` and `runStrict` do the same thing except for some slightly different result handling. Could be merged together a bit.

Reply:
Merged them.